RELEASE_NOTES.md

```diff
@@ -0,0 +1,44 @@
+# Performance Optimizations
+## Intel Architecture Processors
```

Comment:
@tprimak, I think we had a bit more improvements for x64. Please review and update this section.